storage_service/locations/models/package.py

```diff
@@ -190,6 +195,20 @@ def full_pointer_file_path(self):
             self.pointer_file_location.full_path, self.pointer_file_path
         )
 
+    @property
+    def name(self):
+        """Return name of package with trailing UUID removed.
+
+        For AIPs created directly from a transfer, this will be
+        equivalent to the post-sanitization Transfer name.
+        """
+        full_name = Path(self.current_path).name
+        known_extensions = utils.get_compression_extensions()
+        for suffix in full_name.suffixes:
+            if suffix in known_extensions:
+                full_name.replace(suffix, '')
+        return full_name[:-37]
```

Comment:
CI isn't running (will look into that later), but there is something wrong with this code. `suffixes` is a property of the Path object, `name` returns a string. Also `replace` does not mutate, it returns a new string so you need an assignment.

But I get the idea! Thanks. I'd definitely write a test for this with a bunch of different samples. You may find one case where it doesn't work very well, e.g.:

    /var/test/dataset.with.7z.files-a081c05e-ccc2-4bb6-a201-9562a261050a.tar.bz2

I'd expect `dataset.with.7z.files` instead of `dataset.with.files`.

I'm not sure if it's worth the effort, but here's an alternative:

```python
    @property
    def name(self):
        path = Path(self.current_path)
        name, ext = path.name, ''
        for (i, suffix) in reversed(list(enumerate(path.suffixes))):
            if suffix not in KNOWN_EXTENSIONS:
                ext = ''.join(path.suffixes[i+1:])
                break
        return name.replace(ext, '')[:-37]
```

Reply:
Thanks @sevein! Yeah, this is still very much a work in progress. I pushed the changes so far so I can work across computers but will definitely fix, add tests, and take your notes here into account. I'll let you know when it's ready - probably Monday, as I have some conflicting obligations today.

Reply:
Hi @sevein - coming back to this for a moment. I think the code snippet above has the same problem for some paths. E.g. for `/var/test/dataset.with.7z.files-a081c05e-ccc2-4bb6-a201-9562a261050a.7z`, the `.replace(ext, '')` will remove both `.7z` substrings, including the one that's not really a file extension and we want to keep.

I think a solution that avoids this issue might be something like:

```python
path = Path(self.current_path)
name, chars_to_remove = path.name, 37
for suffix in reversed(path.suffixes):
    if suffix not in KNOWN_EXTENSIONS:
        break
    chars_to_remove += len(suffix)
return name[:-chars_to_remove]
```

I'll make time Monday to properly test and make /push the appropriate changes, but any thoughts you have are welcome!

Reply:
Nice! Perhaps I should have used rstrip(ext)?
But your is great! Thanks! 👏